docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+
+The same assets can be bound to different resources and IO managers in different environments. For example, for local development, you might want to store assets on your local filesystem while, in production, you might want to store the assets in S3.
+
+```python file=/concepts/assets/asset_io_manager_prod_local.py startafter=start_marker endbefore=end_marker
+from dagster import AssetGroup, asset, fs_asset_io_manager
+from dagster_aws.s3 import s3_pickle_asset_io_manager, s3_resource
+
+
+@asset
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+
+
+prod_asset_group = AssetGroup(
+    [upstream_asset, downstream_asset],
+    resource_defs={"io_manager": s3_pickle_asset_io_manager, "s3": s3_resource},
+)
+
+local_asset_group = AssetGroup(
+    [upstream_asset, downstream_asset], resource_defs={"io_manager": fs_asset_io_manager}
+)
+```
+
+## Asset jobs
+
+You can define a job that materializes a fixed selection of the assets in an <PyObject object="AssetGroup" /> each time it runs.
+
+The following snippet shows examples three different jobs you might create out of an <PyObject object="AssetGroup" />.
+
+```python file=/concepts/assets/build_job.py startafter=start_marker endbefore=end_marker
+from dagster import AssetGroup, asset
+
+
+@asset
+def upstream():
+    return [1, 2, 3]
+
+
+@asset
+def downstream_1(upstream):
+    return upstream + [4]
+
+
+@asset
+def downstream_2(upstream):
+    return len(upstream)
+
+
+asset_group = AssetGroup([upstream, downstream_1, downstream_2])
+
+all_assets = asset_group.build_job(name="my_asset_job")
+
+downstream_assets = asset_group.build_job(
+    name="my_asset_job", selection=["upstream", "downstream_1"]
+)
+
+upstream_and_downstream_1 = asset_group.build_job(name="my_asset_job", selection="*downstream_1")
+```
+
+Multiple jobs can target overlapping sets of assets.
+
+Like regular jobs, asset jobs can be placed on [schedules](/concepts/partitions-schedules-sensors/schedules) and [sensors](/concepts/partitions-schedules-sensors/sensors).
+
+## Examples
+
+### Multi-component asset keys
+
+Assets are often objects in systems with hierarchical namespaces, like filesystems. Because of this, it often makes sense for an asset key to be a list of strings, instead of just a single string.
+
+```python file=/concepts/assets/multi_component_asset_key.py
+# pylint: disable=redefined-outer-name
+# start_marker
+from dagster import AssetIn, asset
+
+
+@asset(namespace=["one", "two", "three"])
+def upstream_asset():
+    return [1, 2, 3]
+
+
+@asset(ins={"upstream_asset": AssetIn(namespace=["one", "two", "three"])})
+def downstream_asset(upstream_asset):
+    return upstream_asset + [4]
+
+
+# end_marker
+```
+
+### Explicit dependencies
```

Comment:
I think it would also be nice here to include a small section for `non_argument_deps`.

Reply:
I think that's pretty important, but I'm going to leave that for 0.14.1 because we're in overtime already.